File: Data Structures & Algorithms/two-integer-sum/submission-0.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer,Integer> map = new HashMap<>(); 
        int[] solution = {0,0};
        for(int i = 0; i < nums.length; i++){ 
            if(map.containsKey(target-nums[i])){ 
                solution[0] = map.get(target-nums[i]); 
                solution[1] = i;
            }

            map.put(nums[i], i);
        } 
        return solution;
    }
}

File: Data Structures & Algorithms/search-2d-matrix/submission-0.py
class Solution:
    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:
        u, d = 0, len(matrix)-1 
        while u <= d: 
            r = (u+d)//2
            if matrix[r][0] <= target <= matrix[r][-1]: 
                break 
            elif matrix[r][0] > target: 
                d = r -1
            else: 
                u = r + 1
        l, ri = 0, len(matrix[r])-1 
        while l <= ri: 
            x = (ri+l)//2 
            if matrix[r][x] > target: 
                ri = x -1
            elif matrix[r][x] < target: 
                l = x +1
            else: 
                return True 
        return False